toothpick-runtime/src/main/java/toothpick/Configuration.java

```diff
@@ -4,100 +4,182 @@
 import java.util.ArrayList;
 import java.util.LinkedHashSet;
 import toothpick.config.Binding;
+import toothpick.registries.FactoryRegistryLocator;
+import toothpick.registries.MemberInjectorRegistryLocator;
 
 import static java.lang.String.format;
 
+/**
+ * Strategy pattern that allows to change various behaviors of Toothpick.
+ * The default configuration is {@link #production()}.
+ * A custom configuration can be created and used by toothpick,
+ * it is even possible to use a composition of the built-in configurations.
+ */
 public abstract class Configuration {
 
-  /*package-private.*/ static Configuration instance;
+  public static Configuration instance = production();
```

Comment:
we can't make it package private anymore as it is used internally from other packages...